imap/jmap_mailbox.c

```diff
@@ -4158,6 +4158,20 @@ static int _mbox_changes_cb(const mbentry_t *mbentry, void *rock)
         *(data->only_counts_changed) = 0;
     }
 
+    /* Determine where to report that update. Note that we even report
+     * hidden mailboxes in order to allow clients remove unshared ando
+     * deleted mailboxes */
+    json_t *dest = NULL;
+    if ((mbentry->mbtype & MBTYPE_DELETED) || !jmap_hasrights_mbentry(req, mbentry, JACL_LOOKUP)) {
+        if (mbentry->createdmodseq <= data->since_modseq)
+            dest = data->destroyed;
+    } else {
+        if (mbentry->createdmodseq <= data->since_modseq)
```

Comment:
Nit: Can't this if/else be moved up to be else if/else ?

Reply:
thanks. I simplified that if expression.